### PR TITLE
JC-2079 Fix topic modification date

### PR DESCRIPTION
**Pull request by NCNecros**
Add recalculation date after post delete

### CODE REVIEW

File: jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java [hunk 1]
```diff
@@ -200,6 +200,7 @@ public void addPost(Post post) {
      * @param postToRemove post to remove
      */
     public void removePost(Post postToRemove) {
+        recalculateModificationDate();
```
Comment (mixas27) [inline jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java]: `recalculateModificationDate()` is called in `TransactionalPostService#deletePost` we should not duplicate this call. You should investigate why this method was not called in `TransactionalPostService#deletePost` and fix.

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java [hunk 2]
```diff
@@ -69,31 +69,25 @@ public void addPostShouldUpdateModificationDate() throws InterruptedException {
...
     @Test
-    public void updateModificationDateShouldChangeTheModificationDate() {
+    public void addPostShouldSetModificationDateToPostCreationDate() throws InterruptedException {
         Topic topic = createTopic();
         DateTime prevDate = topic.getModificationDate();
-
-        DateTime modDate = topic.updateModificationDate();
-
-        assertNotSame(modDate, prevDate);
+        Thread.sleep(25);
```
Comment (mixas27) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java]: I think, `Thread.sleep()` not a best choise in this case. You can set date manually and use methods like `date.plus Days()` to change it.

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java [hunk 3]
```diff
@@ -69,31 +69,25 @@ public void addPostShouldUpdateModificationDate() throws InterruptedException {
         Topic topic = createTopic();
         DateTime prevDate = topic.getModificationDate();
         Thread.sleep(25); // millisecond precise is a kind of fiction
-        topic.addPost(new Post());
+        Post post = new Post();
+        post.setCreationDate(new DateTime());
```
Comment (mixas27) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java]: Is it neccessary here? Same operation peformed in constructor.

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java [hunk 4]
```diff
@@ -192,6 +185,24 @@ public void removePostShouldRemovePostFromTheTopic() {
...
+        post1.setCreationDate(new DateTime());
+        Post post2 = new Post();
+        post2.setCreationDate(new DateTime());
+        Topic topic = new Topic(new JCUser(), "title");
+        topic.addPost(post1);
+        topic.addPost(post2);
+        Post post3 = new Post();
+        post3.setCreationDate(new DateTime());
+        topic.addPost(post3);
+        assertEquals(topic.getModificationDate(), post3.getCreationDate());
```
Comment (mixas27) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java]: This is case from another method. You should test only one case in test method.

**Comment by ctapobep** [comment]
Merged!

**Pull request closed without merging**